### ISSUE TITLE
Questions about padding mask.

### ISSUE THREAD

**Issue by CadmusTJU** [lucidrains/transfusion-pytorch#45]
Hi! Thanks a lot for your awesome work and contributions to the community! I'm pretty new to multi-modal models, and I was just wondering—why doesn’t the current implementation mask out padding tokens? For example, in forward_test:

`       text = text.masked_fill(text == -1, 0)
        tokens = self.text_embed(text)

        # rotary

        seq_len = tokens.shape[-2]
        pos = torch.arange(seq_len, device = device)

        rotary_emb = self.rotary_emb(pos)

        # attention

        embed, kv_cache = self.transformer(
            tokens,
            rotary_emb = rotary_emb,
            causal_mask = True,
            cache = cache,
            return_kv_cache = True
        )

        # text unembedding

        logits = self.to_text_logits(embed)

        if not return_loss:
            if not return_kv_cache:
                return logits

            return logits, kv_cache

        logits = logits.masked_fill(~self.text_only_logits_mask, max_neg_value(logits))

        loss = F.cross_entropy(
            rearrange(logits, 'b n l -> b l n'),
            labels,
            ignore_index = self.ignore_index
        )
`
It looks like the logits are masked on the padding tokens, but doesn’t that still let the attention attend to them? The attention score calculation would still get affected by the padding tokens.
